packages/protocol/analysisManager.ts

```diff
@@ -160,6 +160,9 @@ class AnalysisManager {
         !handler.onAnalysisResult || client.Analysis.runAnalysis({ analysisId }, sessionId),
         !handler.onAnalysisPoints || client.Analysis.findAnalysisPoints({ analysisId }, sessionId),
       ]);
+    } catch (e) {
+      // log the error if the analysis fails
+      console.error(e);
```

Comment:
Shouldn't this be calling `onAnalysisError` somehow? Otherwise won't it just hang forever?

Reply:
Not sure, i think `onAnalysisError` is reserved for the response failing, as opposed to the request not being created... subtle difference.

Reply:
Hm, interesting. So what's the expected behavior from the POV of the thing calling the API? It will just await the result and then it will need to handle the fact that neither complete nor error were called?

Reply:
Honestly, I gave this 10 minutes of thought earlier. My sole goal was don't crash. We should probably take another stab tomorrow

Reply:
If the analysis fails, I think we should call `onAnalysisError`, regardless of what the error is.